player_rating/tests.py

```diff
@@ -0,0 +1,103 @@
+from .models import PlayerRating, Rating
+from player.models import BallGame, Player
+from django.core.exceptions import ValidationError
+from django.db import IntegrityError
+from django.contrib.auth import get_user_model
+import pytest
+
+
+@pytest.mark.django_db
+class TestPlayerModel:
+
+    @pytest.fixture
+    def user_fixture(self):
+        return get_user_model().objects.create_user(
+            username='testuser',
+            password='testpass'
+        )
+
+    @pytest.fixture
+    def player_fixture(self, user_fixture):
+        player = Player.objects.create(
+            user=user_fixture,
+            birth_date='1990-01-01',
+            favorite_ball_game=BallGame.Soccer
+        )
+        return player
+
+    @pytest.fixture
+    def player_rating_fixture(self, player_fixture):
+        player_rating_fixture = PlayerRating.objects.create(
+            ball_game=BallGame.Soccer,
+            user_id=player_fixture,
+            rating=Rating.Five
+        )
+        return player_rating_fixture
+
+    def test_get_player_rating(self, player_rating_fixture):
+        player_rating_from_db = PlayerRating.objects.get(
+            pk=player_rating_fixture.pk)
+        assert player_rating_from_db == player_rating_fixture
+
+    def test_update_player_rating(self, player_rating_fixture):
+        player_rating_fixture.rating = Rating.Seven
+        player_rating_fixture.save()
+        updated_player_rating = PlayerRating.objects.get(
+            ball_game=BallGame.Soccer,
+            user_id=player_rating_fixture.user_id)
+        assert updated_player_rating.rating == Rating.Seven
+
+    def test_delete_player_rating(self, player_rating_fixture):
+        player_rating_fixture.delete()
+        with pytest.raises(PlayerRating.DoesNotExist):
+            PlayerRating.objects.get(
+                ball_game=BallGame.Soccer,
+                user_id=player_rating_fixture.user_id)
+
+    def test_create_player_rating_invalid_ball_game(self, player_rating_fixture):
+        with pytest.raises(ValidationError):
+            PlayerRating.objects.create(
+                ball_game='Invalid Ball Game',
+                user_id=player_rating_fixture.user_id,
+                rating=Rating.Five).full_clean()
+
+    def test_create_player_rating_invalid_user_id(self):
+        with pytest.raises(IntegrityError):
```

Comment:
Can you check if this is raised by the call for full_clean or earlier? Try removing the call for full_clean and see if it's still passes. If it does, make sure you remove this from all tests that expects `IntegrityError`s.

Reply:
You are correct, the test indeed passes without the full_clean()